### PR TITLE
[Merged by Bors] - feat: port LinearAlgebra.StdBasis

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/StdBasis.lean [hunk 1]
```diff
@@ -112,44 +112,41 @@ theorem infᵢ_ker_proj_le_supᵢ_range_stdBasis {I : Finset ι} {J : Set ι} (h
...
 #align linear_map.supr_range_std_basis_eq_infi_ker_proj LinearMap.supᵢ_range_stdBasis_eq_infᵢ_ker_proj
 
 theorem supᵢ_range_stdBasis [Finite ι] : (⨆ i, range (stdBasis R φ i)) = ⊤ := by
   cases nonempty_fintype ι
-  convert top_unique (infi_emptyset.ge.trans <| infi_ker_proj_le_supr_range_std_basis R φ _)
-  ·
-    exact
-      funext fun i =>
-        ((@supᵢ_pos _ _ _ fun h => range <| std_basis R φ i) <| Finset.mem_univ i).symm
+  convert top_unique (infᵢ_emptyset.ge.trans <| infᵢ_ker_proj_le_supᵢ_range_stdBasis R φ _)
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/StdBasis.lean:138]: I think `convert` might have syntax for naming the variables it intros now.

File: Mathlib/LinearAlgebra/StdBasis.lean [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    (⨅ i ∈ J, ker (proj i : (∀ i, φ i) →ₗ[R] φ i)) ≤ ⨆ i ∈ I, range (stdBasis R φ i) :=
+  SetLike.le_def.2
+    (by
+      intro b hb
+      simp only [mem_infᵢ, mem_ker, proj_apply] at hb
+      rw [←
+        show (∑ i in I, stdBasis R φ i (b i)) = b by
+          ext i
+          rw [Finset.sum_apply, ← stdBasis_same R φ i (b i)]
+          refine' Finset.sum_eq_single i (fun j hjI ne => stdBasis_ne _ _ _ _ ne.symm _) _
```
Comment (eric-wieser) [inline Mathlib/LinearAlgebra/StdBasis.lean:120]: One thing that might help here is using `refine` instead and replacing only the relevant `_` with `?_`

Comment (Ruben-VandeVelde) [reply Mathlib/LinearAlgebra/StdBasis.lean:120]: What would this help with?

File: Mathlib/LinearAlgebra/StdBasis.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply Mathlib/LinearAlgebra/StdBasis.lean:138]: Not in the documentation. Let me know what it is if you find it.

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**